Distinguish different document types for applications

BertalanD opened the pull request:
- Introduce a type enum in the files table
- Add a dedicated section in the UI for each required document type
- Add validation rules that check if each required document type was
  provided

The description texts in the UI can be changed via the configurable
texts framework without re-deploying the code.

Work towards #742

Sorry for the painful Hunglish, but I'm not going to try to translate the official document type names to English.

BertalanD committed:
Distinguish different document types for applications

- Introduce a type enum in the files table
- Add a dedicated section in the UI for each required document type
- Add validation rules that check if each required document type was
  provided

The description texts in the UI can be changed via the configurable
texts framework without re-deploying the code.

A few small drive-by changes are included:
- Mandate minimum length for the motivational/cover letter question
- Fix paragraph breaks in long-form questions not showing up on the
  admissions UI

BertalanD committed:
Move the profile selector to personal information for applications

horcsinbalint commented on app/Http/Controllers/Secretariat/UserController.php:
```diff
@@ -69,7 +69,7 @@ public function storeProfilePicture(Request $request, User $user): RedirectRespo
             Storage::delete($old_profile->path);
         } else {
-            $user->profilePicture()->create(['path' => $path, 'name' => 'profile_picture']);
+            $user->profilePicture()->create(['path' => $path, 'type' => 'profile_picture']);
```
AI's suggestion is not wrong, but does not really matter

horcsinbalint commented on app/Models/Application.php:
```diff
@@ -413,13 +423,37 @@ public function missingData(): array
+            case FileType::RESUME:
+                return true;
+            case FileType::BESOROLASI_HATAROZAT:
+                return $this->user->educationalInformation->studyLines()->where('start', '=', Semester::next()->id)->exists();
```
I think we should probably use `app(\App\Http\Controllers\Auth\ApplicationController::class)->semester` for getting the current semester as the application is a periodic event

horcsinbalint commented on app/Utils/ApplicationHandler.php:
```diff
@@ -70,9 +71,10 @@ public function storeFile(Request $request, $user): void
         ]);
         $path = $request->file('file')->store('uploads');
-        $user->application->files()->create(['path' => $path, 'name' => $request->input('name')]);
+        $user->application->files()->create(['path' => $path, 'type' => $request->input('type'), 'description' => $request->input('name')]);
```
Just a question: is it safe to allow users to upload any type of document (like a `receipt`)?

BertalanD replied:
Hm, I don't think that has any security implications. They will still have their `application` foreign key set, so will be visible in the admission UI and will be deleted after the admissions process finishes.

horcsinbalint commented on resources/views/auth/application/application.blade.php:
```diff
@@ -306,7 +306,7 @@ class="hideable-picture"
+                                                   target="_blank">{{ $file->description }}</a>
                                             </div>
                                         </div>
                                     @empty
```
I think line #317 is no longer necessary.

horcsinbalint reviewed (commented):
Thank you, the changes seem to be quite good, I added some nitpicking comments

BertalanD commented on resources/sass/materialize.scss:
```diff
@@ -390,3 +390,11 @@ body.dark a.btn {
+.form-disabled {
+    opacity: 0.5;
+
+    & .collapsilble-body {
```
I'm just removing that. We should still be able to interact with the dropdown and the file upload/delete buttons.

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in f7101be according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/752

BertalanD committed:
Update admin file upload functionality for typed documents

BertalanD committed:
Only allow PDF files for CVs and "besorolási határozat"

These should always be digital documents, so we shouldn't accommodate
pictures.

BertalanD committed:
Warn before navigating away from unsaved long-form application question

BertalanD committed:
Pre-fill document names for CV and besorolási határozat
